rizinorg/rizin
# Fix plugin removal and clean code

**Your checklist for this pull request**

**Detailed description**



- when removing a plugin through `L-` the "plugins" list in RzCore/RzDemangler/RzAsm/etc. was not updated. Define a _dt callback in the plugins handler for each type to make sure that list is updated
- clean the lib.c code and document it
- move dlopen/dlsym/dlclose to sys.c

**Test plan**



- compile an external plugin
- load it
- check it works
- unload it
- check it is not present anymore
